actionpack/lib/action_dispatch/testing/test_process.rb

```diff
@@ -17,7 +17,24 @@ module FixtureFile
       def fixture_file_upload(path, mime_type = nil, binary = false)
         if self.class.respond_to?(:fixture_path) && self.class.fixture_path &&
             !File.exist?(path)
-          path = File.join(self.class.fixture_path, path)
+          original_path = path
+          path = Pathname.new(File.join(self.class.fixture_path, path))
+
+          if !self.class.file_fixture_path
+            ActiveSupport::Deprecation.warn(<<~EOM)
+              `fixture_file_upload` now expects a path relative to the `file_fixture_path` which is
+              currently not set. Please set `file_fixture_path`.
+            EOM
+          elsif path.exist?
+            non_deprecated_path = path.relative_path_from(self.class.file_fixture_path)
+            ActiveSupport::Deprecation.warn(<<~EOM)
+              `fixture_file_upload` now expects a path relative to the `file_fixture_path`.
+              Please modify the call from
+              `fixture_file_upload(#{original_path})` to `fixture_file_upload(#{non_deprecated_path})`.
+            EOM
+          else
+            path = file_fixture(original_path)
```

Comment:
This is within `self.class.respond_to?(:fixture_path) && self.class.fixture_path` so we won't use `file_fixture_path` without `fixture_path` being defined. 

Maybe this makes more sense?
```
unless File.exist?(path)
  if self.class.respond_to?(:fixture_path) && self.class.fixture_path
    path_from_fixture_path = Pathname.new(self.class.fixture_path).join(path)
    if  path_from_fixture_path.exist?
      non_deprecated_path = path.relative_path_from(self.class.file_fixture_path)
      ActiveSupport::Deprecation.warn(<<~EOM)
        `fixture_file_upload` now expects a path relative to the `file_fixture_path`.
        Please modify the call from
        `fixture_file_upload(#{path})` to `fixture_file_upload(#{non_deprecated_path})`.
      EOM
      path = non_deprecated_path
    end
  end
  path = file_fixture(path)
end
Rack::Test::UploadedFile.new(path, mime_type, binary)
```